orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/rollback/ExplicitRollback.groovy

```diff
@@ -124,10 +127,73 @@ class ExplicitRollback implements Rollback {
     }
 
     stages << disableServerGroupStage
-    stages << buildApplySourceServerGroupCapacityStage(parentStage, parentStage.mapTo(ResizeStrategy.Source))
+
+    // only restore the min if it was pinned, i.e. there was a resize
+    if (resizeStage != null) {
+      stages << buildApplySourceServerGroupCapacityStage(parentStage, parentStage.mapTo(ResizeStrategy.Source))
+    }
     return stages
   }
 
+  @Nullable TargetServerGroup lookupServerGroup(Stage parentStage, String serverGroupName) {
```

Comment:
nit: Wouldn't mind seeing `@Nonnull` etc added to the method args, too.

Reply:
do we have an explicit convention around this or tooling that actually takes advantage of that?

Otherwise it seems overkill, for me the implicit contract is that things that are not explicitly `@Nullable` must be `@Nonnull`